core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -121,180 +121,130 @@ public TableOperations underlyingOps() {
     return ops;
   }
 
-  private void checkLastOperationCommitted(String operation) {
+  protected <T extends PendingUpdate> T appendUpdate(T update) {
```

Comment:
I am not strongly opinionated on this but I'm wondering if we should also make this final

Reply:
I can think of some might want its own inner transaction like manageSnapshots for overridng this. But add final for now seems reasonable. We can always change it later once the needs come